src/helpers/git_helpers/branch_configuration_helpers.sh

```diff
@@ -104,9 +103,53 @@ function ensure_knows_parent_branches {
       else
         # here we don't know the parent of the current branch -> ask the user
         echo
-        echo -n "Please enter the parent branch for $(echo_inline_cyan_bold "$current_branch") ($(echo_inline_dim "$MAIN_BRANCH_NAME")): "
+        local branches=$(git for-each-ref --sort=-committerdate refs/heads/ --format='%(refname:short)')
+
+        function print_branch {
+          local number=$1
+          local branch_name=$2
+
+          output_style_bold
+          printf "%3s: " "$number"
+          output_style_reset
+          echo "$branch_name"
+        }
+
+        echo
+        echo "Feature branches can be branched directly off "
+        echo "$MAIN_BRANCH_NAME or from other feature branches."
+        echo
+        echo "The former allows to develop and ship features completely"
+        echo "independently of each other, the latter to work on new things"
+        echo "that require other changes currently under review."
+        echo
+
+        local branch_numbers
+        print_branch 1 "$MAIN_BRANCH_NAME"
+        branch_numbers[1]=$MAIN_BRANCH_NAME
+        i=2
+        for branch in $branches; do
+          if [ "$branch" != "$current_branch" -a "$branch" != "$MAIN_BRANCH_NAME" ]; then
+            branch_numbers[i]=$branch
+            print_branch $i "$branch"
+            i=$(( i + 1 ))
+          fi
+        done
```

Comment:
I think we should try to break this function up as it has grown quite large. Mentioned it on another PR so it can be dealt with separately.